File: src/models/SubmitResultVO.ts
/**
 * Submission result model
 */
export class SubmitResultVO {
  /** Status code: 1(Success), 21(Already exists), 22(Queued), other(Error) */
  code: number;

  /** Description */
  description: string;

  /** Task ID */
  result?: string;

  /** Extended fields */
  properties: Record<string, any> = {};

  constructor(code: number, description: string, result?: string) {
    this.code = code;
    this.description = description;
    this.result = result;
  }

  /**
   * Set property
   */
  setProperty(name: string, value: any): this {
    this.properties[name] = value;
    return this;
  }

  /**
   * Remove property
   */
  removeProperty(name: string): this {
    delete this.properties[name];
    return this;
  }

  /**
   * Get property
   */
  getProperty(name: string): any {
    return this.properties[name];
  }

  /**
   * Create success result
   */
  static of(code: number, description: string, result?: string): SubmitResultVO {
    return new SubmitResultVO(code, description, result);
  }

  /**
   * Create fail result
   */
  static fail(code: number, description: string): SubmitResultVO {
    return new SubmitResultVO(code, description);
  }
}

File: src/main/java/com/github/novicezk/midjourney/result/SubmitResultVO.java
package com.github.novicezk.midjourney.result;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;

import java.util.HashMap;
import java.util.Map;

@Data
@ApiModel("Submission Result")
public class SubmitResultVO {

	@ApiModelProperty(value = "Status code: 1(Success), 21(Already exists), 22(Queued), other(Error)", required = true, example = "1")
	private int code;

	@ApiModelProperty(value = "Description", required = true, example = "Success")
	private String description;

	@ApiModelProperty(value = "Task ID", example = "1320098173412546")
	private String result;

	@ApiModelProperty(value = "Extended fields")
	private Map<String, Object> properties = new HashMap<>();

	public SubmitResultVO setProperty(String name, Object value) {
		this.properties.put(name, value);
		return this;
	}

	public SubmitResultVO removeProperty(String name) {
		this.properties.remove(name);
		return this;
	}

	public Object getProperty(String name) {
		return this.properties.get(name);
	}

	@SuppressWarnings("unchecked")
	public <T> T getPropertyGeneric(String name) {
		return (T) getProperty(name);
	}

	public <T> T getProperty(String name, Class<T> clz) {
		return clz.cast(getProperty(name));
	}

	public static SubmitResultVO of(int code, String description, String result) {
		return new SubmitResultVO(code, description, result);
	}

	public static SubmitResultVO fail(int code, String description) {
		return new SubmitResultVO(code, description, null);
	}

	private SubmitResultVO(int code, String description, String result) {
		this.code = code;
		this.description = description;
		this.result = result;
	}
}
